src/Mod/PartDesign/Gui/TaskExtrudeParameters.cpp

```diff
@@ -681,16 +681,18 @@ void TaskExtrudeParameters::getReferenceAxis(App::DocumentObject*& obj, std::vec
     }
 }
 
-void TaskExtrudeParameters::onButtonFace(const bool pressed)
+void TaskExtrudeParameters::onButtonFace(const bool checked)
 {
+    if (!checked && ui->lineFaceName->text().isEmpty())
+        ui->lineFaceName->setPlaceholderText(tr("No face selected"));
+    else if (checked && ui->lineFaceName->text().isEmpty())

```

Comment:
Sounds good, but note that it is possible to just type the face "name" here.

Reply:
You can do this already:
- select a face
- change its name

The solution is simple in my opinion: make the line edit read-only because it is not meant to be that the user changes there something on his own.

I did this now.

Reply:
I was just referring to the "select a face" placeholder. Making the linedit readonly is one way to add consistency.

Reply:
since you are setting a placeholder text, do you really need to check if the lineedit is empty? once you have text the placeholder text shouldn't be displayed anymore.

Reply:
What do you mean? The placeholder is only shown if there is now text. That is the idea of the placeholder.

Reply:
that's my point, qt already only shows the palceholder when there is no text, so why would you need to check if it's empty? It doesn't harm I guess but I don't understand why the check is there.

Reply:
Ah, this is indeed no longer necessary. I think in my initial PR it was different and later I did not review it thoroughly.
I fixed this now and also simplified the code a bat: https://github.com/FreeCAD/FreeCAD/commit/98dae1b2